backend/services/create_team.py

```diff
@@ -0,0 +1,52 @@
+from battle.battles.battle import validate_sum_pokemons, verify_pokemon_is_saved
+
+from pokemon.helpers import verify_pokemon_exists_api
+
+
+def verify_if_data_is_valid(data):  # pylint: disable=too-many-return-statements
+    message_error = None
+
+    if 'battle' not in data:
+        message_error = 'ERROR: Select a valid battle'
+        return message_error
+
+    if data['trainer'] not in (data['battle'].creator, data['battle'].opponent):
+        message_error = ("ERROR: You do not have permission for this action.")
+        return message_error
```

Comment:
This validation on DRF should be a permission:  https://www.django-rest-framework.org/api-guide/permissions/#custom-permissions

Reply:
For the form, this should be a mixin, you can use the LoginRequired mixin as example: https://docs.djangoproject.com/en/3.2/topics/auth/default/#the-loginrequired-mixin

Reply:
thanks for all the support :dancers: